streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -220,8 +245,18 @@ public StreamsConfig(Map<?, ?> props) {
         super(CONFIG, props);
     }
 
-    public Map<String, Object> getConsumerConfigs(StreamThread streamThread, String groupId, String clientId) {
-        Map<String, Object> props = getBaseConsumerConfigs();
+    public Map<String, Object> getConsumerConfigs(StreamThread streamThread, String groupId, String clientId) throws ConfigException {
+        // first disable auto commit ignoring any user overridden values,
+        // this is necessary for streams commit semantics
```

Comment:
This comment needs to be updated, we now fail instead of disabling and ignoring.